src/main/java/org/terasology/logic/inventory/ItemCommands.java

```diff
@@ -170,4 +263,146 @@ public String bulkGiveItem(
         }
         return result;
     }
+
+
+    public String removeBlock(
+            @Sender EntityRef sender,
+            @CommandParam("blockName") String uri,
+            @CommandParam(value = "quantity", required = false) Integer quantityParam,
+            @CommandParam(value = "shapeName", required = false) String shapeUriParam) {
+        Set<ResourceUrn> matchingUris = Assets.resolveAssetUri(uri, BlockFamilyDefinition.class);
+
+        BlockFamily blockFamily = null;
+
+        if (matchingUris.size() == 1) {
+            Optional<BlockFamilyDefinition> def = Assets.get(matchingUris.iterator().next(), BlockFamilyDefinition.class);
+            if (def.isPresent()) {
+                if (def.get().isFreeform()) {
+                    if (shapeUriParam == null) {
+                        blockFamily = blockManager.getBlockFamily(new BlockUri(def.get().getUrn(), new ResourceUrn("engine:cube")));
+                    } else {
+                        Set<ResourceUrn> resolvedShapeUris = Assets.resolveAssetUri(shapeUriParam, BlockShape.class);
+                        if (resolvedShapeUris.isEmpty()) {
+                            return "Found block. No shape found for '" + shapeUriParam + "'";
+                        } else if (resolvedShapeUris.size() > 1) {
+                            StringBuilder builder = new StringBuilder();
+                            builder.append("Found block. Non-unique shape name, possible matches: ");
+                            Iterator<ResourceUrn> shapeUris = sortItems(resolvedShapeUris).iterator();
```

Comment:
You could also restructure this and replace it with the method below, as both are `Set<ResourceUrn>`